src/prefect/settings/models/_defaults.py

```diff
@@ -80,7 +80,6 @@ def default_database_connection_url(settings: "Settings") -> SecretStr:
             "host",
             "user",
             "name",
-            "password",
```

Comment:
@desertaxle  If user intend to IAM role based login, they don't need to setup password initially. But here it's required. 

I am not sure removing `password` from required is the best approach here. 

Do you have any suggestion to better way to handle it?

Reply:
This doesn't engage if a user sets the full database URL, and we have handling further down if the `password` setting isn't set, so I think this change is fine.